fuel-p2p/src/gossipsub/messages.rs

```diff
@@ -0,0 +1,24 @@
+use serde::{Deserialize, Serialize};
+use std::io;
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+pub enum GossipsubMessage {
+    BroadcastNewTx,
+    BroadcastNewBlock,
+}
+
+impl GossipsubMessage {
```

Comment:
I think we should consider a trait-based approach here so we could configure different codecs besides bincode in the future.

Reply:
Agreed there, I don't think we can encapsulate here all the variants that will be trafficked in the network.

We should, instead, attempt to parse per domain. In the case of consensus events, it would be something that encapsulates this struct: https://github.com/FuelLabs/fuel-bft/blob/bb6b7dc5ad66905ad9d15c8a1fe35329bc37e770/src/vote.rs#L10-L17

Reply:
@vlopes11 please refer to https://fuellabs.slack.com/archives/C01L1HUJSL8/p1650294323378019 
the correct messages will be defined in another PR since I was blocked by #258 (the core-interfaces)

Reply:
Perfect! Do we have a tracking issue for that?

Reply:
@vlopes11 I will create one, as a matter of fact the PR will come in asap as well

Reply:
@Voxelot I have made those functions generic so that they accept any kind of Deserializer and Serializer